schedview/app/prenight/prenight.py

```diff
@@ -1,418 +1,530 @@
-import panel as pn
+import param
 import logging
-from copy import deepcopy
-import pickle
-import lzma
-from tempfile import TemporaryDirectory, NamedTemporaryFile
+import pandas as pd
+import os
 
-import numpy as np
 from astropy.time import Time
 
-import rubin_sim
 from rubin_sim.scheduler.model_observatory import ModelObservatory
 import rubin_sim.scheduler.example
-from rubin_sim.scheduler.utils import SchemaConverter
 
 import schedview.compute.astro
 import schedview.collect.opsim
 import schedview.compute.scheduler
 import schedview.collect.footprint
+import schedview.plot.visits
 import schedview.plot.visitmap
 import schedview.plot.rewards
 import schedview.plot.visits
 import schedview.plot.maf
 import schedview.plot.nightbf
 
-TEMP_DIR = TemporaryDirectory()
-DEFAULT_TIMEZONE = "Chile/Continental"
+import panel as pn
 
-pn.extension("tabulator", css_files=[pn.io.resources.CSS_URLS["font-awesome"]])
+AVAILABLE_TIMEZONES = [
+    "Chile/Continental",
+    "US/Pacific",
+    "US/Arizona",
+    "US/Mountain",
+    "US/Central",
+    "US/Eastern",
+]
+DEFAULT_TIMEZONE = AVAILABLE_TIMEZONES[0]
+DEFAULT_CURRENT_TIME = Time.now()
+DEFAULT_OPSIM_FNAME = "opsim.db"
+DEFAULT_SCHEDULER_FNAME = "scheduler.pickle.xz"
+DEFAULT_REWARDS_FNAME = "rewards.h5"
+USE_EXAMPLE_SCHEDULER = False
+
+
+pn.extension(
+    "tabulator",
+    css_files=[pn.io.resources.CSS_URLS["font-awesome"]],
+    sizing_mode="stretch_width",
+)
+pn.config.console_output = "disable"
+
+logging.basicConfig(format="%(asctime)s %(message)s", level=logging.INFO)
+
+debug_info = pn.widgets.Debugger(
+    name="Debugger info level", level=logging.DEBUG, sizing_mode="stretch_both"
+)
+
+
+class Prenight(param.Parameterized):
+
+    opsim_output_fname = param.String(DEFAULT_OPSIM_FNAME)
```

Comment:
Could use param.Filename here maybe .. result would show up in report with full pathname, which may be useful if multiple directories have the same opsim file?

Reply:
I considered param.Filename, but it won't work, because we want to support loading data from URLs, not just files on the user's laptop.